src/hotspot/share/gc/g1/g1FullGCPrepareTask.cpp

```diff
@@ -119,6 +125,20 @@ void G1FullGCPrepareTask::G1CalculatePointersClosure::free_humongous_region(Heap
   dummy_free_list.remove_all();
 }
 
+void G1FullGCPrepareTask::G1CalculatePointersClosure::free_pinned_region(HeapRegion* hr) {
```

Comment:
Should this be called free_archive_region (or free_open_archive_region)?  The statistics counter is `_pinned_archive_regions_removed`, so this presumably can't be used for some other kind of pinned region.